kratos/python_scripts/analysis_stage.py

```diff
@@ -280,3 +281,19 @@ def __CreateListOfProcesses(self):
         order_processes_initialization = self._GetOrderOfOutputProcessesInitialization()
         self._list_of_output_processes = self._CreateProcesses("output_processes", order_processes_initialization)
         self._list_of_processes.extend(self._list_of_output_processes) # Adding the output processes to the regular processes
+
+    def __CheckIfSolveSolutionStepReturnsAValue(self, is_converged):
+        """In case the solver does not return the state of convergence
+        (same as the SolvingStrategy does) then issue ONCE a deprecation-warning
+        """
+        if is_converged is None:
+            if not hasattr(self, '_map_ret_val_depr_warnings'):
+                self._map_ret_val_depr_warnings = []
+            solver_class_name = self._GetSolver().__class__.__name__
+            # used to only print the deprecation-warning once
+            if not solver_class_name in self._map_ret_val_depr_warnings:
+                self._map_ret_val_depr_warnings.append(solver_class_name)
+                from KratosMultiphysics.kratos_utilities import IssueDeprecationWarning
+                warn_msg  = 'Solver "{}" does not return '.format(solver_class_name)
```

Comment:
@ddiezrod I just realized that I again used the new syntax for formatting the string.

Please let me know if this gives problems again. 
Sorry for the inconvenience

(FYI also @maceligueta since you also use python 2)

Reply:
@philbucher This works correctly in python 2.7, the problem in the other case was the int conversion not the .format which can be safely used in python 2.7 (I think).